### PR TITLE
[SYCL][Doc] Update docs to reflect PI removal.

### LINKED ISSUE #14928: review all documentation pertaining to PI and either remove it, update it or re-direct it to point at docs in the UR repo - mentioned here

### PR DESCRIPTION
**Pull request by aarongreig**
Fixes #14928

### CODE REVIEW

**Review by dm-vodopyanov — COMMENTED** [review]
Also these doxygen docs have to be updated:

1
https://github.com/intel/llvm/blob/07bf3c11ebeafbbb408708dd79f578ec7c974423/sycl/source/detail/context_impl.hpp#L61-L70

2
https://github.com/intel/llvm/blob/07bf3c11ebeafbbb408708dd79f578ec7c974423/sycl/source/detail/context_impl.hpp#L105-L112

3
https://github.com/intel/llvm/blob/07bf3c11ebeafbbb408708dd79f578ec7c974423/sycl/source/detail/context_impl.hpp#L115-L122

4
https://github.com/intel/llvm/blob/07bf3c11ebeafbbb408708dd79f578ec7c974423/sycl/source/detail/event_impl.hpp#L61-L67

5
https://github.com/intel/llvm/blob/07bf3c11ebeafbbb408708dd79f578ec7c974423/sycl/source/detail/event_impl.hpp#L129-L137

6
https://github.com/intel/llvm/blob/07bf3c11ebeafbbb408708dd79f578ec7c974423/sycl/source/detail/kernel_impl.hpp#L34-L41

7
https://github.com/intel/llvm/blob/07bf3c11ebeafbbb408708dd79f578ec7c974423/sycl/source/detail/platform_impl.hpp#L35-L39

8
https://github.com/intel/llvm/blob/07bf3c11ebeafbbb408708dd79f578ec7c974423/sycl/source/detail/platform_impl.hpp#L107-L113

**Comment by dm-vodopyanov** [comment]
Also other things:

9
https://github.com/intel/llvm/blob/07bf3c11ebeafbbb408708dd79f578ec7c974423/.github/CODEOWNERS#L46

has to be updated, we don't have `sycl/pi_win_proxy_loader` anymore.

10
https://github.com/intel/llvm/blob/07bf3c11ebeafbbb408708dd79f578ec7c974423/.github/CODEOWNERS#L125-L131
We need to rename @intel/dpcpp-nativecpu-pi-reviewers as well (to @intel/dpcpp-nativecpu-ur-reviewers ?) @stdale-intel, can you please rename the group? @aarongreig or @stdale-intel, can you please update the CODEOWNERS file after that?

**Comment by dm-vodopyanov** [comment]
\+ more

11
https://github.com/intel/llvm/tree/sycl/sycl/test-e2e/Plugin folder has to be renamed? (+ rename in CODEOWNERS after that)

12
https://github.com/intel/llvm/blob/07bf3c11ebeafbbb408708dd79f578ec7c974423/.github/workflows/sycl-linux-build.yml#L172

13
https://github.com/intel/llvm/blob/07bf3c11ebeafbbb408708dd79f578ec7c974423/.github/workflows/sycl-macos-build-and-test.yml#L56

14
https://github.com/intel/llvm/blob/07bf3c11ebeafbbb408708dd79f578ec7c974423/buildbot/configure.py#L54

15
https://github.com/intel/llvm/blob/07bf3c11ebeafbbb408708dd79f578ec7c974423/buildbot/configure.py#L95

16
https://github.com/intel/llvm/blob/07bf3c11ebeafbbb408708dd79f578ec7c974423/buildbot/configure.py#L179

17
https://github.com/intel/llvm/blob/07bf3c11ebeafbbb408708dd79f578ec7c974423/buildbot/configure.py#L71

18
https://github.com/intel/llvm/blob/07bf3c11ebeafbbb408708dd79f578ec7c974423/buildbot/configure.py#L155-L156

**Comment by dm-vodopyanov** [comment]
\+ update XPTI

19
https://github.com/intel/llvm/blob/07bf3c11ebeafbbb408708dd79f578ec7c974423/xpti/doc/SYCL_Tracing_Implementation.md?plain=1#L66

20
https://github.com/intel/llvm/blob/07bf3c11ebeafbbb408708dd79f578ec7c974423/sycl/source/detail/xpti_registry.hpp#L30

21
https://github.com/intel/llvm/blob/07bf3c11ebeafbbb408708dd79f578ec7c974423/sycl/unittests/xpti_trace/NodeCreation.cpp#L45

22
https://github.com/intel/llvm/blob/07bf3c11ebeafbbb408708dd79f578ec7c974423/sycl/unittests/xpti_trace/QueueIDCheck.cpp#L49

(@KseniyaTikhomirova can provide more information about the update of XPTI (if any)).

**Comment by dm-vodopyanov** [comment]
29
https://github.com/intel/llvm/blob/07bf3c11ebeafbbb408708dd79f578ec7c974423/CONTRIBUTING.md?plain=1#L61

**Comment by dm-vodopyanov** [comment]
32
https://github.com/intel/llvm/blob/07bf3c11ebeafbbb408708dd79f578ec7c974423/sycl/cmake/modules/AddSYCL.cmake#L47-L98

**Comment by aarongreig** [comment]
@dm-vodopyanov thanks for all the feedback

For point #1, the comment is still correct as it refers to the Plugin object (which arguably should be renamed but that isn't a task for this PR)
For points #14, #15 and #16, the cmake variable is still called that. I think I'll spin out a separate PR to rename it.
Points #30 and #31 are kind of weird, we still call them "plugin releases" on our website (the page pointed to). That's an issue I can raise internally, but it might be worth leaving the wording as is to match up with what you see if you click the link.

I'm not sure what to do about #23, #24, #25 and #26. The pi_collector example will for sure need re-written now that sycl.pi has been removed, I'll open another sub-ticket on the post-merge work issue to look into this as it's not as trivial as just updating docs.

File: sycl/doc/design/images/RuntimeArchitecture-with-fusion.svg
Comment (aarongreig) [reply sycl/doc/design/images/RuntimeArchitecture-with-fusion.svg:1]: fixed

File: sycl/doc/design/images/RuntimeArchitecture.svg
Comment (aarongreig) [reply sycl/doc/design/images/RuntimeArchitecture.svg:1]: fixed

**Comment by aarongreig** [comment]
> Points #30 and #31 are kind of weird, we still call them "plugin releases" on our website (the page pointed to). That's an issue I can raise internally, but it might be worth leaving the wording as is to match up with what you see if you click the link.

It isn't clear if we'll change the terminology on the website, but it for sure wont be until after the next release since the current latest release on there is inarguably a PI plugin.

File: sycl/doc/design/KernelFusionJIT.md
Comment (sommerlukas) [inline sycl/doc/design/KernelFusionJIT.md:1]: I think we should also update or remove the diagrams from this file which still mention PI.

Comment (aarongreig) [reply sycl/doc/design/KernelFusionJIT.md:1]: I think I've updated them all now

**Review by sommerlukas — APPROVED** [review]
Changes to kernel fusion documentation LGTM.

**Review by PietroGhg — APPROVED** [review]
Native CPU LGTM, thank you

**Comment by dm-vodopyanov** [comment]
> Also other things:
> 
> 9
> 
> https://github.com/intel/llvm/blob/07bf3c11ebeafbbb408708dd79f578ec7c974423/.github/CODEOWNERS#L46
> 
> has to be updated, we don't have `sycl/pi_win_proxy_loader` anymore.
> 
> 10
> 
> https://github.com/intel/llvm/blob/07bf3c11ebeafbbb408708dd79f578ec7c974423/.github/CODEOWNERS#L125-L131
> 
> 
> We need to rename @intel/dpcpp-nativecpu-pi-reviewers as well (to @intel/dpcpp-nativecpu-ur-reviewers ?) @stdale-intel, can you please rename the group? @aarongreig or @stdale-intel, can you please update the CODEOWNERS file after that?

Adding @stdale-intel to reviewers because of the action above

File: xpti/doc/SYCL_Tracing_Implementation.md
Comment (bader) [inline xpti/doc/SYCL_Tracing_Implementation.md:66]: Do you plan to rename `initializePlugins` to something like `initializeBackends`?

File: sycl/source/detail/kernel_impl.hpp
Comment (bader) [inline sycl/source/detail/kernel_impl.hpp:36]: Does it make sense to say `UR adapter` instead of `adapter` for more clarity?

File: sycl/doc/design/UnifiedRuntime.md
Comment (bader) [inline sycl/doc/design/UnifiedRuntime.md:9]: Will the name `Plugin` be changed in the future?

Comment (aarongreig) [reply sycl/doc/design/UnifiedRuntime.md:9]: it probably would make sense to divorce it from the old terminology, I'm thinking possible alternatives could be `Backend` or maybe `Adapter` - what do you think?

File: xpti/doc/SYCL_Tracing_Implementation.md
Comment (aarongreig) [reply xpti/doc/SYCL_Tracing_Implementation.md:66]: I think that should come in with the "rename `Plugin`" change, right now the function technically is initializing a list of `Plugin` objects

File: sycl/source/detail/kernel_impl.hpp
Comment (aarongreig) [reply sycl/source/detail/kernel_impl.hpp]: agreed that does read a bit better

File: sycl/doc/design/UnifiedRuntime.md
Comment (bader) [reply sycl/doc/design/UnifiedRuntime.md:9]: Agree. My understanding is that `Backend` is good alternative for SYCL Runtime plug-ins, whereas `Adapter` is a for Unified Runtime plug-ins.

Comment (aarongreig) [reply sycl/doc/design/UnifiedRuntime.md:9]: this was quite a noisy change so I spun it out into its own PR https://github.com/intel/llvm/pull/15363

**Comment by aarongreig** [comment]
> > Also other things:
> > 9
> > https://github.com/intel/llvm/blob/07bf3c11ebeafbbb408708dd79f578ec7c974423/.github/CODEOWNERS#L46
> > 
> > has to be updated, we don't have `sycl/pi_win_proxy_loader` anymore.
> > 10
> > https://github.com/intel/llvm/blob/07bf3c11ebeafbbb408708dd79f578ec7c974423/.github/CODEOWNERS#L125-L131
> > 
> > We need to rename @intel/dpcpp-nativecpu-pi-reviewers as well (to @intel/dpcpp-nativecpu-ur-reviewers ?) @stdale-intel, can you please rename the group? @aarongreig or @stdale-intel, can you please update the CODEOWNERS file after that?
> 
> Adding @stdale-intel to reviewers because of the action above

maybe we can spin this out into its own issue?

**Review by Bensuo — APPROVED** [review]

File: sycl/doc/design/SYCLInstrumentationUsingXPTI.md
Comment (kbenzie) [inline sycl/doc/design/SYCLInstrumentationUsingXPTI.md:320]: This file contains conflict markers

**Review by kbenzie — CHANGES_REQUESTED** [review]

**Comment by aarongreig** [comment]
ping @intel/dpcpp-specification-reviewers @intel/dpcpp-devops-reviewers

**Review by gmlueck — APPROVED** [review]
Reviewing just the specification change ... the change to "sycl_ext_oneapi_discard_queue_events.asciidoc" is OK.

**Review by kbenzie — APPROVED** [review]

**Comment by aarongreig** [comment]
@steffenlarsen would you mind taking a look at this so we can get it over the line?

**Comment by aarongreig** [comment]
> > > Also other things:
> > > 9
> > > https://github.com/intel/llvm/blob/07bf3c11ebeafbbb408708dd79f578ec7c974423/.github/CODEOWNERS#L46
> > > 
> > > has to be updated, we don't have `sycl/pi_win_proxy_loader` anymore.
> > > 10
> > > https://github.com/intel/llvm/blob/07bf3c11ebeafbbb408708dd79f578ec7c974423/.github/CODEOWNERS#L125-L131
> > > 
> > > We need to rename @intel/dpcpp-nativecpu-pi-reviewers as well (to @intel/dpcpp-nativecpu-ur-reviewers ?) @stdale-intel, can you please rename the group? @aarongreig or @stdale-intel, can you please update the CODEOWNERS file after that?
> > 
> > 
> > Adding @stdale-intel to reviewers because of the action above
> 
> maybe we can spin this out into its own issue?

@dm-vodopyanov @stdale-intel I've created an issue to track so we can unblock merging this https://github.com/intel/llvm/issues/15956

File: sycl/doc/design/SYCLInstrumentationUsingXPTI.md
Comment (steffenlarsen) [inline sycl/doc/design/SYCLInstrumentationUsingXPTI.md:240]: Woops!
```suggestion
```

**Review by steffenlarsen — APPROVED** [review]
Looks reasonable! Just one small mistake. 👍

**Review by bader — APPROVED** [review]
Please, resolve merge conflicts.

**Comment by aarongreig** [comment]
@intel/llvm-gatekeepers I think we're good to merge this

**Pull request merged**
